plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -228,6 +233,54 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        int length = toType.getLength().orElse(Integer.MAX_VALUE);
+        if (fromType.isShort()) {
+            return new ShortDecimalToVarcharCoercer(fromType, length);
+        }
+        return new LongDecimalToVarcharCoercer(fromType, length);
+    }
+
+    private static class ShortDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        private final int currentLength;
+
+        protected ShortDecimalToVarcharCoercer(DecimalType fromType, int currentLength)
+        {
+            super(fromType, VARCHAR);
+            this.currentLength = currentLength;
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            BigDecimal bigDecimal = BigDecimal.valueOf(fromType.getLong(block, position), fromType.getScale());
+            String stringValue = bigDecimal.toPlainString();
+            toType.writeString(blockBuilder, stringValue.substring(0, min(currentLength, stringValue.length())));
```

Comment:
This can truncate digits also before the decimal dot.
Is it what Hive does?

For now, I'd prefer to throw on overflow. Perhaps also when fraction is being truncated.
I hope people don't change eg `decimal(8, x)` to `varchar(6)` columns.

See also https://github.com/trinodb/trino/pull/14179

Reply:
I checked and this is what hive is doing. I can change it but I thought we want to do the same as hive

Reply:
In principle -- yes, we want to follow Hive. Especially for the set of convertible type pairs.
I feel uneasiness when we copy what I perceive as bugs.
I do hope we don't need this weird truncation here to make real-life use-cases work?

In any case, add a comment explaining what's the Hive behavior.